src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -0,0 +1,256 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Formats.Cbor;
+
+namespace System.Security.Cryptography.Cose
+{
+    public class CoseHeaderMap : IEnumerable<(CoseHeaderLabel, ReadOnlyMemory<byte>)>
+    {
+        private static readonly byte[] s_emptyBstrEncoded = new byte[] { 0x40 };
+        private bool _isReadOnly;
+        public bool IsReadOnly { get => _isReadOnly; internal set => _isReadOnly = value; }
```

Comment:
This can just be:
```C#
public bool IsReadOnly { get; internal set; }
```
and the code can reference IsReadOnly rather than _isReadOnly.